cmd/k8sm-executor/main.go

```diff
@@ -54,135 +48,120 @@ var (
-}
-
-func getApiserverClient() (*client.Client, error) {
-	clientConfig := client.Config{}
-	if *authPath != "" {
-		authInfo, err := clientauth.LoadFromFile(*authPath)
-		if err != nil {
-			return nil, err
-		}
-		clientConfig, err = authInfo.MergeWithConfig(clientConfig)
-		if err != nil {
-			return nil, err
-		}
-	}
-	// TODO: adapt Kube client to support LB over several servers
-	if len(apiServerList) > 1 {
-		log.Infof("Mulitple api servers specified.  Picking first one")
-	}
-	clientConfig.Host = apiServerList[0]
-	if c, err := client.New(&clientConfig); err != nil {
-		return nil, err
-	} else {
-		return c, nil
-	}
+	flag.Var(&clusterDNS, "cluster_dns", "IP address for a cluster DNS server.  If set, kubelet will configure all containers to use this for DNS resolution in addition to the host's DNS servers")
 }
 
 func main() {
 
 	flag.Parse()
-	util.InitLogs() // TODO(jdef) figure out where this actually sends logs by default
+	util.InitLogs()
 	defer util.FlushLogs()
 	rand.Seed(time.Now().UTC().UnixNano())
 
 	verflag.PrintAndExitIfRequested()
 
-	etcd.SetLogger(util.NewLogger("etcd "))
-
-	// Make an API client if possible.
-	if len(apiServerList) < 1 {
-		log.Info("No api servers specified.")
-	} else {
-		if apiClient, err := getApiserverClient(); err != nil {
-			log.Errorf("Unable to make apiserver client: %v", err)
-		} else {
-			// Send events to APIserver if there is a client.
-			log.Infof("Sending events to APIserver.")
-			record.StartRecording(apiClient.Events(""), "kubelet")
-		}
+	if err := util.ApplyOomScoreAdj(*oomScoreAdj); err != nil {
+		log.Info(err)
+	}
+
+	kcfg := standalone.KubeletConfig{
+		Address:                 address,
+		AuthPath:                *authPath,
+		ApiServerList:           apiServerList,
+		AllowPrivileged:         *allowPrivileged,
+		HostnameOverride:        *hostnameOverride,
+		RootDirectory:           *rootDirectory,
+		NetworkContainerImage:   *networkContainerImage,
+		SyncFrequency:           *syncFrequency,
+		RegistryPullQPS:         *registryPullQPS,
+		RegistryBurst:           *registryBurst,
+		MinimumGCAge:            *minimumGCAge,
+		MaxContainerCount:       *maxContainerCount,
+		ClusterDomain:           *clusterDomain,
+		ClusterDNS:              clusterDNS,
+		Port:                    *port, // TODO(nnielsen): Don't hardwire port, but use port from resource offer.
+		CAdvisorPort:            *cAdvisorPort,
+		EnableServer:            true,
+		EnableDebuggingHandlers: *enableDebuggingHandlers,
+		DockerClient:            util.ConnectToDockerOrDie(*dockerEndpoint),
+		EtcdClient:              kubelet.EtcdClientOrDie(etcdServerList, *etcdConfigFile),
 	}
 
-	// Log the events locally too.
-	record.StartLogging(log.Infof)
+	driver := new(mesos.MesosExecutorDriver)
 
-	capabilities.Initialize(capabilities.Capabilities{
-		AllowPrivileged: *allowPrivileged,
-	})
+	//HACK(jdef) largely hacked from the k8s standalone package, createAndInitKubelet
```

Comment:
Did you mean to leave this //HACK comment in? Should it be replaced with an appropriate TODO or an @see reference?